File: FizzBuzz/FizzBuzzPlayer.cs
using System;

namespace FizzBuzz
{
    public class FizzBuzzPlayer
    {
        private ILogger logger;

        public FizzBuzzPlayer(ILogger logger)
        {
            this.logger = logger;
        }

        public string play(int input)
        {
            var word = Word(input);
            var result = word == "" ? input.ToString() : word;
            logger.Log($"The FizzBuzz of {input} is {result}");
            return result;
        }

        private static string Word(int input)
        {
            var result = "";
            if (input % 3 == 0)
                result += "Fizz";
            if (input % 5 == 0)
                result += "Buzz";
            return result;
        }
    }
}

File: FizzBuzz/FizzBuzzPlayer.js
function FizzBuzzPlayer() {
    const logFormat = "The FizzBuzz of {input} is {output}."

    return {
         play: play
        // play: (value) => (value % 3 == 0 ? "Fizz" : "") + (value % 5 == 0 ? "Buzz" : "") == "" ? value.toString() : (value % 3 == 0 ? "Fizz" : "") + (value % 5 == 0 ? "Buzz" : "")
    };

    function play(value) {       
        var returnValue = "";
        
        returnValue += value % 3 == 0 ? "Fizz" : "";
        returnValue += value % 5 == 0 ? "Buzz" : "";
        returnValue = returnValue == "" ? value.toString() : returnValue;
        log(value, returnValue);
        return returnValue;
    }

    function log(input, output) {
        console.log(
            logFormat
            .replace('{input}', input)
            .replace('{output}', output)
            )
    }
}